src/convert-package.ts

```diff
@@ -86,51 +117,33 @@ export function configureConverter(
 }
 
 /**
- * Converts an entire package from HTML imports to JS modules
+ * Convert a package-layout project to JavaScript modules & npm.
  */
-export async function convertPackage(options: ConvertPackageOptions) {
-  const outDir = options.outDir || 'js_out';
-  const outDirResolved = path.resolve(process.cwd(), outDir);
-  console.log(`Out directory: ${outDirResolved}`);
-
+export default async function convert(options: ConvertPackageOptions) {
+  const outDir = options.outDir;
   const npmPackageName = options.packageName;
   const npmPackageVersion = options.packageVersion;
+  console.log(`Out directory: ${outDir}`);
 
+  const bowerJson = readJson(options.inDir, 'bower.json');
   const analyzer = configureAnalyzer(options);
   const analysis = await analyzer.analyzePackage();
   const converter = configureConverter(analysis, options);
   const results = await converter.convert();
-  if (options.cleanOutDir) {
-    rimraf.sync(outDirResolved);
-  }
-
-  try {
-    await fs.mkdir(outDirResolved);
-  } catch (e) {
-    if (e.errno === -17) {  // directory exists
-      // do nothing
-    } else {
-      throw e;
-    }
-  }
+  await setupOutDir(options.outDir, !!options.cleanOutDir);
 
   for (const [newPath, newSource] of results) {
-    const outPath = path.resolve(outDirResolved, newPath);
-    mkdirp.sync(path.dirname(outPath));
-    if (newSource !== undefined) {
-      await fs.writeFile(outPath, newSource);
-    } else if (fs.existsSync(outPath)) {
-      await fs.unlink(outPath);
-    }
+    const filePath = path.resolve(outDir, newPath);
+    await writeFile(filePath, newSource);
```

Comment:
If you wanted to try to get some parallelism here (I'm curious if fs writes even benefit), you could use Promise.all:

```ts
await Promise.all(IterableX.from(results).map(([newPath, newSource]) => 
    writeFile(path.resolve(outDir, newPath), newSource)));
```

Reply:
Not sure if Node can do parallel writes but it can at least queue them up at I/O level more efficiently. This should have an even bigger effect on convert-workspace.